src/lib.rs

```diff
@@ -173,6 +173,26 @@ impl<T> Cactus<T> {
     pub fn vals(&self) -> CactusValsIter<T> {
         CactusValsIter{next: self.node.as_ref()}
     }
+
+    /// Consume this Cactus node and return its data. If the cactus node has no children, its data
+    /// is returned without cloning; if the node has children then the internal data is cloned
+    /// (hence why `T` must implement the `Clone` trait).
+    ///
+    /// # Examples
+    /// ```
+    /// use cactus::Cactus;
+    /// let c = Cactus::new().child(1).child(2);
+    /// let p = c.parent().unwrap();
+    /// assert_eq!(c.take_or_clone_val(), Some(2));
+    /// // At this point c has been consumed and can no longer be referenced.
+    /// assert_eq!(p.val(), Some(&1));
+    /// ```
+    pub fn take_or_clone_val(self) -> Option<T> where T: Clone {
```

Comment:
You can create a `Cactus` with something that isn't `Clone`able, but you won't be able to call `take_or_clone_val` on it. So this compiles OK:

```rust
        struct S { woo: usize }
        let c = Cactus::new().child(S{woo: 2});
```

but if you add this line:

```
        c.take_or_clone_val();
```

you get a complaint like:

```
328 |         c.take_or_clone_val();
    |           ^^^^^^^^^^^^^^^^^ the trait `std::clone::Clone` is not implemented for `tests::test_take_or_clone_val::S`
```